src/main/java/reposense/parser/CsvParser.java

```diff
@@ -223,17 +226,34 @@ private String getRowContentAsRawString(final CSVRecord record) {
      * Generates map of column header to position number for input {@code possibleHeader}.
      * @throws InvalidCsvException if {@code possibleHeader} does not contain all the mandatory headers.
      */
-    private void validateHeader(String[] possibleHeader) throws InvalidCsvException {
+    private void validateHeader(String[] possibleHeader) throws InvalidCsvException, InvalidHeaderException {
         int headerSize = possibleHeader.length;
+        Set<String> knownColumns = new HashSet<>();
+        StringBuilder unknownColumns = new StringBuilder();
```

Comment:
I understand you are trying to build the string for the final output here, but I think it's better practice to store as a string array first, then use `String.join()` at the end when throwing the InvalidHeaderException.